builder/sources/registry/manifest.go

```diff
@@ -79,14 +83,30 @@ func (registry *Registry) ManifestV2(repository, reference string) (*manifestV2.
 		return nil, err
 	}
 
+	logrus.Infof("[ManifestV2] body received %s", string(body))
+
 	deserialized := &manifestV2.DeserializedManifest{}
 	err = deserialized.UnmarshalJSON(body)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unmarshal JSON: %v", err)
+	}
+	return nil, nil
+}
+
+// CheckManifest checks if the manifest of the given image is exist.
+func (registry *Registry) CheckManifest(repository, reference string) error {
```

Comment:
The new function replacement the function ManifestV2 only removes code about DeserializedManifest.
so, you can change the function ManifestV2.

Reply:
ok